File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Java/flight/Schedule.java
package flight;

import java.time.LocalDateTime;

public class Schedule {
    private LocalDateTime departureTime;
    private LocalDateTime arrivalTime;

    public Schedule(LocalDateTime departureTime, LocalDateTime arrivalTime) {
        this.departureTime = departureTime;
        this.arrivalTime = arrivalTime;
    }

    public LocalDateTime getDepartureTime() {
        return departureTime;
    }

    public LocalDateTime getArrivalTime() {
        return arrivalTime;
    }
}

File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Cpp/flight/src/Schedule.cpp
#include "../include/Schedule.h"

Schedule::Schedule(const std::chrono::system_clock::time_point& departureTime,
                   const std::chrono::system_clock::time_point& arrivalTime)
    : departureTime(departureTime), arrivalTime(arrivalTime) {}

const std::chrono::system_clock::time_point& Schedule::getDepartureTime() const {
    return departureTime;
}

const std::chrono::system_clock::time_point& Schedule::getArrivalTime() const {
    return arrivalTime;
}
